aidial_adapter_openai/utils/prompt_truncate.py

```diff
@@ -0,0 +1,65 @@
+from typing import Callable, List, Set, Tuple, TypeVar
+
+from aidial_sdk.exceptions import (
+    TruncatePromptSystemAndLastUserError,
+    TruncatePromptSystemError,
+)
+
+_T = TypeVar("_T")
+
+DiscardedMessages = List[int]
+UsedTokens = int
+
+
+def truncate_prompt(
+    message_holders: List[_T],
+    message_tokens_getter: Callable[[_T], int],
+    is_system_message_getter: Callable[[_T], bool],
+    max_prompt_tokens: int,
+    initial_prompt_tokens: int = 3,
```

Comment:
The default here is strange - why is 3 the best default value for this **generic algorithm**? 3 comes from OpenAI tokenizer, I would have passed it explicitly.

Reply:
fixed